### ISSUE TITLE
False positive for Jinja variables regex (test provided + failing CI output)

### ISSUE THREAD

**Issue by myii** [warpnet/salt-lint#283]
**Describe the bug**

Encountered a false positive for the Jinja variables check, based on this block:

```sls
  file.blockreplace:
    - ...
    - content: |
           DKIM_DOMAIN = ${lc:${domain:$h_from:}}
           DKIM_KEY_FILE = /etc/exim4/dkim/{{ pillar['dkim_private_key'] }}
           DKIM_PRIVATE_KEY = ${if exists{DKIM_KEY_FILE}{DKIM_KEY_FILE}{0}}
           DKIM_SELECTOR = {{ pillar['dkim_selector'] }}
```

With the error being:

```sls
[206] Jinja variables should have spaces before and after: '{{ var_name }}'
.../init.sls:71
           DKIM_DOMAIN = ${lc:${domain:$h_from:}}
```

**To Reproduce**

I've provided a failing test here:

* https://github.com/myii/salt-lint/commit/df975f167341b3a24d3d3fce6c124f9b03dcdb09

As I've mentioned there, it fails within quotes as well:

```sls
    - content: 'DKIM_DOMAIN = ${lc:${domain:$h_from:}}'
```

Here's the failure in GitHub Actions based on that commit:

* https://github.com/myii/salt-lint/actions/runs/2441054851

Note, there's also another commit there, which adjusts the CI so that the `yamllint` error is resolved and the CI doesn't fail fast (so that all instance failures can be seen):

* https://github.com/myii/salt-lint/commit/3b5c7a3fed5f569a656b0852f99d275a766350e3

Let me know if you'd like that in a PR here.

**Expected behavior**

The regex shouldn't catch this as a failure.
